### DISCUSSION TITLE
Convert Unet to SNN

### DISCUSSION THREAD

**Discussion by cnrgbot** [nengo/nengo-zoo#919] [Deep Learning]
> _Archived from the Nengo Discourse forum._ Originally posted by **ntquyen2** on 2021-04-24. (Forum topic #1698, post #1 / id 5153)

I convert Unet model to SNN to increase the accuracy but the problem occurs. I don’t know what is the reason when I train the period model with 4 milion parameters such as VGG or Alexnet ( even so more 4 milion), no problem but my UNET with 1.809 milion parameter, the crash (overload RAM) occured. Can you help me, please? This is my code (UNET model)? And I see tf.keras.layers.Cropping2D and tf.keras.layers.ConvolutionTransform2D can not convert to native Nengo Object so the accuracy is affected, is’n it?
[UNET_SNN_Nengo.py|attachment](https://raw.githubusercontent.com/nengo/forum-archive/main/uploads/default/original/1X/0e160f5842efff4a007bc3db50d0cbc9f2fd4001.py) (5.2 KB)

**Comment by cnrgbot** [comment]
> _Archived from the Nengo Discourse forum._ Originally posted by **Eric** on 2021-04-26. (Forum topic #1698, post #2 / id 5163)

While TensorFlow memory is related to the number of parameters, there are also other things that use TensorFlow memory, such that a model with fewer parameters doesn't necessarily use less memory.

Probably the easiest thing to try is using a smaller minibatch size. It looks like you're using 200. You could try 128, and if that doesn't work try 64, 32, 16, 8, etc. until you get one that works. (I prefer powers of 2 because they often run faster on GPUs.)

The fact that some layers can't be converted to native Nengo objects won't necessarily be detrimental to accuracy in NengoDL. One main downside is that it may make it impossible to run the network on other backends, e.g. NengoLoihi.

**Reply by cnrgbot** [reply]
> _Archived from the Nengo Discourse forum._ Originally posted by **ntquyen2** on 2021-04-26. (Forum topic #1698, post #3 / id 5164)

Thanks for your response. Can you answer my second question that is **tf.keras.layers.Cropping2D and tf.keras.layers.ConvolutionTransform2D can not convert to native Nengo Object so the accuracy is affected, is’n it?**

**Reply by cnrgbot** [reply]
> _Archived from the Nengo Discourse forum._ Originally posted by **Eric** on 2021-04-26. (Forum topic #1698, post #4 / id 5165)

I did: 

> *Originally @​Eric wrote [post #2, topic #1698]:*
>
> The fact that some layers can’t be converted to native Nengo objects won’t necessarily be detrimental to accuracy in NengoDL. One main downside is that it may make it impossible to run the network on other backends, e.g. NengoLoihi.
